IronFoundry.Warden/Utilities/EnvironmentBlock.cs

```diff
@@ -0,0 +1,136 @@
+﻿
+
+namespace IronFoundry.Warden.Utilities
+{
+    using System;
+    using System.Collections;
+    using System.Collections.Generic;
+    using System.Runtime.InteropServices;
+    using System.Text;
+
+    public class EnvironmentBlock
+    {
+        private readonly Dictionary<string,string> _environment = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+
+        public static EnvironmentBlock GenerateDefault()
+        {
+            var defaultEnvBlock = CreateDefaultEnvBlock();
+            var environment = new EnvironmentBlock();
+            environment.Merge(defaultEnvBlock);
+
+            return environment;
+        }
+
+        public static EnvironmentBlock Create(IDictionary dictionary)
+        {
+            Dictionary<string, string> typedDictionary = new Dictionary<string, string>();
+
+            foreach (var key in dictionary.Keys)
+            {
+                typedDictionary[key.ToString()] = dictionary[key].ToString();
+            }
+
+            var envBlock = new EnvironmentBlock().Merge(typedDictionary);
+
+            return envBlock;
+        }
+
+
+        /// <summary>
+        /// Upserts the specified environment variables into the existing ones.
+        /// </summary>
+        public EnvironmentBlock Merge(IDictionary<string, string> environmentVariables)
+        {
+            foreach (var kv in environmentVariables)
+            {
+                _environment[kv.Key] = kv.Value;
+            }
+
+            return this;
+        }
+
+        public EnvironmentBlock Merge(EnvironmentBlock envBlock)
+        {
+            return this.Merge(envBlock._environment);
+        }
+
+
+        public IDictionary<string, string> ToDictionary()
+        {
+            return new Dictionary<string, string>(_environment);
```

Comment:
Do we need to specify the comparer again? Or will it inherit the case-insensitive comparer?